### PR TITLE
Replace use of Castle.DynamicProxy for actor code with SourceGenerator

### PR DESCRIPTION
**Pull request by alexperovich**
https://github.com/dotnet/arcade/issues/11539

Sample generated code: [PullRequestActor.g.cs](https://gist.github.com/alexperovich/3cb04c34f449792b89021899d1208b20)

### CODE REVIEW

**Comment by garath** [comment]
What does all this help with?

**Review by MattGal — DISMISSED** [review]

**Comment by alexperovich** [comment]
The current method using DynamicProxy is difficult to grok and messes with stack traces when debugging stuff. The generated code is easy to see and less crazy to understand. Also there was an issue I was seeing in logs where the dynamic proxy stuff was failing to intercept something it should have, this fixes that problem also.

File: src/Microsoft.DotNet.ServiceFabric.ServiceHost.Analyzers/ServiceHostGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                {
+                    Expression: MemberAccessExpressionSyntax
+                    {
+                        RawKind: (int) SyntaxKind.SimpleMemberAccessExpression,
+                        Name: GenericNameSyntax
+                        {
+                            Identifier.Text: string name,
+                        },
+                    }
+                } && name == methodName, (syntax, _) => ((InvocationExpressionSyntax)syntax.Node, syntax.SemanticModel));
```
Comment (dkurepa) [inline src/Microsoft.DotNet.ServiceFabric.ServiceHost.Analyzers/ServiceHostGenerator.cs:33]: nit: new line for the (syntax, _) ....

**Comment by tkapin** [comment]
Several questions @alexperovich:
1) What is the concrete reason to implement this now? Is it done as part of FR?
1) What epic is this work being part of? The parent issue #11539 is closed for long time now. 
1) How is the change tested?

To be clear, I'm not opposing the change, but the above is not clear. Also, PRs should be tracked under opened issues.

File: src/Microsoft.DotNet.ServiceFabric.ServiceHost.Analyzers/ServiceHostGenerator.cs (hunk 1 above)
Comment (alexperovich) [reply src/Microsoft.DotNet.ServiceFabric.ServiceHost.Analyzers/ServiceHostGenerator.cs]: fixed.

**Comment by alexperovich** [comment]
> 1. What is the concrete reason to implement this now? Is it done as part of FR?
I was investigating an issue on friday and got fed up with the DynamicProxy stuff being opaque and decided to spend an hour of spare time fixing it.

> 2. What epic is this work being part of? The parent issue #11539 is closed for long time now.
No epic. I just saw a problem, had some spare cycles, and fixed it. This change has no business being in release notes.

> 3. How is the change tested?
I ran the maestro scenario tests locally with no issues.

**Comment by premun** [comment]
I think this has gotten out of sync too much as we've moved the shared libraries mostly out of this repo. I understand there is some work done here and that it might improve things but with Alex out of the team I don't see this being completed.

Feel free to re-open if you feel different.

**Pull request closed without merging**